main/util.py

```diff
@@ -214,16 +216,38 @@ def create_name_from_email(email):
   return re.sub(r'_+|-+|\.+|\++', ' ', email.split('@')[0]).title()
 
 
-def password_hash(user_db, password):
+def password_hash(password, salt):
   m = hashlib.sha256()
-  m.update(user_db.key.urlsafe())
-  m.update(user_db.created.isoformat())
-  m.update(m.hexdigest())
   m.update(password.encode('utf-8'))
-  m.update(config.CONFIG_DB.salt)
+  m.update(salt)
   return m.hexdigest()
 
 
+def password_salt(n_bytes=32):
+  return binascii.hexlify(os.urandom(n_bytes))
+
+
+# Taken from Django Source Code
+# Helps prevent timing attacks
+# See http://carlos.bueno.org/2011/10/timing.html
+def constant_time_compare(val1, val2):
```

Comment:
Regardless of the fact that this PR is not going to make it (as discussed in #294) I don't see any point of doing that here as we are comparing hashes and the hashes always have the same length.. am I missing something?